rewrite-test/src/main/kotlin/org/openrewrite/java/tree/MethodTypeBuilderTest.kt

```diff
@@ -0,0 +1,239 @@
+                }
+                
+                String notification() {
+                    return "You have been notified";
+                }
+            }
+        """.trimIndent())
+        val b = (((a[0].classes[0].body.statements[0] as J.MethodDecl).body.statements[0] as J.VariableDecls).vars[0].initializer.type as JavaType.Method).declaringType
+
+        val m = newMethodType()
+            .name("notification")
+            .declaringClass("A")
+            .build()
+
+        assertThat(m.declaringType).isEqualTo(b)
+    }
+
+    @Test
+    fun methodTypeWithoutReturnType(jp: JavaParser) {
+        val method = newMethodType()
+            .name("notification")
+            .declaringClass("A")
+            .build()
+
+        assertThat(method.resolvedSignature.returnType).isEqualTo(JavaType.Primitive.Void)
+        assertThat(method.genericSignature.returnType).isEqualTo(JavaType.Primitive.Void)
+    }
+
+    @Test
+    fun methodTypeWithReturnType(jp: JavaParser) {
+        val cu = jp.parse("""
+            public class A {
+                public void foo() {
+                    String str1 = notification1();
+                    String[] str2 = notification2();
+                }
+                
+                String notification1() {
+                    return "You have been notified once";
+                }
+                
+                String[] notification2() {
+                    return new String[]{"You have been notified", "Of two important things"};
+                }
+            }
+        """.trimIndent())
+        val expectedReturnType1 = (((cu[0].classes[0].body.statements[0] as J.MethodDecl).body.statements[0] as J.VariableDecls).vars[0].initializer.type as JavaType.Method).resolvedSignature.returnType
+        val expectedReturnType2 = (((cu[0].classes[0].body.statements[0] as J.MethodDecl).body.statements[1] as J.VariableDecls).vars[0].initializer.type as JavaType.Method).resolvedSignature.returnType
+
+        val method1 = newMethodType()
+            .name("notification1")
+            .declaringClass("A")
+            .returnType("java.lang.String")
+            .build()
+
+        val method2 = newMethodType()
+            .name("notification2")
+            .declaringClass("A")
+            .returnType(JavaType.Array(JavaType.buildType("java.lang.String")))
+            .build()
+
+        assertThat(method1.resolvedSignature.returnType).isEqualTo(expectedReturnType1)
+        assertThat(method2.resolvedSignature.returnType).isEqualTo(expectedReturnType2)
+    }
+
+    @Test
+    fun methodTypeWithGenericReturnType(jp: JavaParser) {
+        val cu = jp.parse("""
+            public class A {
+                public void foo() {
+                    String str = notification();
+                }
+                
+                <T> T notification() {
+                    return "You have been notified";
+                }
+            }
```

Comment:
This wouldn't compile. I'll leave it as an exercise for you to determine why and fix it.

You can use gradle to quickly create scratch projects with the `gradle init` command. Or you can use the site Jonathan mentioned yesterday, https://start.spring.io/